# Analysis Part I

Find 95% lower (one-sided) and two-sided confidence intervals for the reduction in risk corresponding to the primary endpoint (data “Through day 29”), using method 3 and also using the cruder conservative approach via simultaneous Bonferroni confidence bounds for N⋅1 and N1⋅ described in the notes on causal inference. (For the Bonferroni approach to two-sided intervals, use Sterne’s method for the underlying hypergeometric confidence intervals. Feel free to re-use your own code from the previous problem set.)


In [1]:
from utils import hypergeom_conf_interval
from math import comb
from scipy.stats import binom, hypergeom
from cabin.cabin import *

In [2]:
n, m = 753, 752
N = n+m

n01 = 59
n11 = 11
n00 = m - n01
n10 = n - n11

alpha = 0.05
print(n11, n10, n01, n00)

(11, 742, 59, 693)

## Method 3

In [ ]:
result = tau_twoside(11, 742, 59, 684, 0.05, 100)
tau_lower = result['tau_lower']
tau_upper = result['tau_lower']

It takes long time to run, here we only shown the results:

tau_upper = -0.0314

tau_lower = -0.0975

## Bonferroni approach

#### Two-sided:

In [7]:
# two sided
N_1 = hypergeom_conf_interval(n, n11, N, 1-alpha/2, alternative='two-sided', method= "sterne")
N1_ = hypergeom_conf_interval(m, n01, N, 1-alpha/2, alternative='two-sided', method= "sterne")

tao_lower = (N_1[0] - N1_[1])/N
tao_upper = (N_1[1] - N1_[0])/N
print("tau_upper", round(tao_upper,4))
print("tau_lower", round(tao_lower,4))

tau_upper -0.0399
tau_lower -0.0877


#### Lower one-side

In [8]:
# lower one sided
N_1 = hypergeom_conf_interval(n, n11*N/n, N, 1-alpha/2, alternative='lower')
N1_ = hypergeom_conf_interval(m, n01*N/m, N, 1-alpha/2, alternative='upper')

tao_lower = (N_1[0] - N1_[1])/N
tao_upper = (N_1[1] - N1_[0])/N
print("tau_upper", round(tao_upper,4))
print("tau_lower", round(tao_lower,4))

tau_upper 1.0
tau_lower -0.1561


#### Upper one-side

In [9]:
# lower one sided
N_1 = hypergeom_conf_interval(n, n11*N/n, N, 1-alpha/2, alternative='upper')
N1_ = hypergeom_conf_interval(m, n01*N/m, N, 1-alpha/2, alternative='lower')

tao_lower = (N_1[0] - N1_[1])/N
tao_upper = (N_1[1] - N1_[0])/N
print("tau_upper", round(tao_upper,4))
print("tau_lower", round(tao_lower,4))

tau_upper -0.101
tau_lower -1.0


## Discuss the differences between the two sets of confidence intervals.

Confidence interval for reduction in risk by using Method 3 is $\tau_{method3} = [ -0.0975, -0.0314]$. And the confidence interval for reduction by using sterne method is $\tau_{sterne} = [-0.0877, -0.0399]$. The sterne's confidence is narrow than the method3 confidence interval. In Sterne's tails, they may be of different sizes, and create a potentially tighter confidence interval than method 3.

## Is it statistically legitimate to use one-sided confidence intervals? Why or why not?

Yes, for this case, if the vaccine is effective, there should be fewer infected cases in the treatment group compared to the placebo group. We only need to check whether $\tau$ is smaller than 0 to find the upper one-sided confidence interval.

## Are the 2-sided confidence intervals preferable to the one-sided intervals? Why or why not?

Yes, the 2-sided confidence interval is preferable to the one-sided interval. Because we can check how far away the lower bound to the -1 to determine the effectiveness of the vaccine. If close to -1, it means that the vaccine is effective and merely any cases after taking the vaccine treatment.
